integration_tests/__init__.py

```diff
@@ -186,3 +186,27 @@ def update_name_and_version(self, project_dir, name=None, version=None):
             else:
                 print(line)
         return updated_project_dir
+
+    def gated(self, snap_name, expected_validations, expected_error=None):
```

Comment:
Roberto, expected_validations has to be optional as well, otherwise you end up passing a and expected result that is completely irrelevant in the testing scenario.